clients/java/src/test/java/io/camunda/client/incident/SearchIncidentTest.java

```diff
@@ -216,6 +217,22 @@ public void shouldConvertIncidentErrorType() {
     }
   }
 
+  @Test
+  void shouldSearchIncidentByResourceNotFoundError() {
+    // when
+    client
+        .newIncidentQuery()
+        .filter(f -> f.incidentKey(1L).errorType(RESOURCE_NOT_FOUND))
+        .send()
+        .join();
+
+    // then
+    final IncidentSearchQuery request = gatewayService.getLastRequest(IncidentSearchQuery.class);
+    final IncidentFilter filter = request.getFilter();
+    assertThat(filter.getIncidentKey()).isEqualTo("1");
+    assertThat(filter.getErrorType()).isEqualTo(ErrorTypeEnum.RESOURCE_NOT_FOUND);
+  }
```

Comment:
🔧 You could consider parameterizing this test using each of the enum values from the engine. That way, it automatically works (or requires changes) when a new enum value is added in the engine.